src/Core/Convert.php

```diff
@@ -474,11 +474,15 @@ public static function bytes2memstring($bytes, $decimal = 0)
     {
         $scales = ['B','K','M','G','T','P','E','Z','Y'];
         // Get scale
-        $scale = (int)floor(log($bytes ?? 0.0, 1024));
+        $value = floor(log($bytes ?? 0.0, 1024));
+        // Handle values such as `-INF` - prior to PHP 8.5 `(int) -INF` yields 0,
+        // from 8.5 onwards it still does, though raises a PHP warning.
+        // https://www.php.net/ChangeLog-8.php
+        // - Casting floats that are not representable as ints now emits a warning.
+        $scale = is_finite($value) ? (int) $value : 0;
```

Comment:
```suggestion
        // Handle infinite values such as `-INF` since that is "not representable as int" and emits a warning.
        $scale = is_finite($value) ? (int) $value : 0;
```

This comment is gonna get stale really quickly. We should avoid saying things like "in php 8.5..." in code comments, and put those in PR comments instead if it's not clear from context.

Reply:
Updated